kratos/utilities/function_parser_utility.cpp

```diff
@@ -24,25 +27,64 @@
 namespace Kratos
 {
 
-// A check only to check that the x is not part of exp
-inline bool CheckThereIsNotx(const std::string& rString)
+/**
+ * @brief Ternary version of adjacent_find, with padding.
+ * @details Scans a sequence with a 3-window, passing it to a predicate.
+ * The window is padded such that every element of the sequence
+ * is the middle element of the window once.
+ * @return Returns the iterator to the middle of the first window to
+ * fulfill the predicate.
+ */
+template<typename ForwardIterator, typename TernaryPredicate>
+ForwardIterator PaddedWindowFind(
+    ForwardIterator Begin,
+    ForwardIterator End,
+    typename std::iterator_traits<ForwardIterator>::value_type Padding,
+    TernaryPredicate&& Pred
+    )
 {
-    const auto first_x = rString.find(std::string("x"));
-    if (first_x != std::string::npos) {
-        const std::string exp_string = "exp";
-        const char e_char = exp_string[0];
-        const char x_char = exp_string[1];
-        const char p_char = exp_string[2];
-        for(std::size_t i = first_x; i < rString.size(); ++i) {
-            if(rString[i] == x_char) {
-                if (!(rString[i - 1] == e_char && rString[i + 1] == p_char)) {
-                    return false;
-                }
-            }
-        }
+    // Edge case: length 0
+    if(Begin == End) {
+        return End;
+    }
+    
+    auto wmiddle = std::next(Begin); // Middle of the window
+    
+    // Edge case: length 1
+    if(wmiddle == End) {
+        return Pred(Padding, *Begin, Padding) ? Begin : End;
+    }
+
+    if(Pred(Padding, *Begin, *wmiddle)) return Begin;
+    
+    auto wlast = std::next(wmiddle); // Last element of the window
+    while(wlast != End) {
+        if(Pred(*Begin, *wmiddle, *wlast)) return wmiddle;
+        std::advance(Begin, 1);
+        std::advance(wmiddle, 1);
+        std::advance(wlast, 1);
     }
+    // wlast points to end, so it is replaced with padding
+    return Pred(*Begin, *wmiddle, Padding) ? wmiddle : End;
+}
 
-    return true;
+/** 
+ * @brief CHECKING IF THE FUNCTION DEPENDS ON SPACE f(var in {x,y,z,X,Y,Z})
+ * @details Appearances of these letters that are not a dependence in space:
+ *  - Variable containing the letter: Starts with a letter, followed by letters, numbers, or underscores
+ *  - Hex number:                     Starts with 0x, followed by numbers or abcdefABCDEF
+ *  - Function containing x:          exp
+ */
+bool ExpressionDependsOnSpace(std::string const& rExpression) {
+    return PaddedWindowFind(rExpression.cbegin(), rExpression.cend(), '\0', 
+        [] (char lead, char middle, char last) {
+            constexpr auto const& variables = "xyzXYZ";
+            if(std::find(std::begin(variables), std::end(variables), middle) == std::end(variables)) {
+                return false; // Middle character is not xyzXYZ
+            }
+            return !std::isalnum(lead) && !std::isalnum(last);
```

Comment:
What about cases with leading digits other than 0? This would reject ```1x``` ... ```9x```, right?

Reply:
Right, I wasn't sure whether tinyexpr supported shorthand factors (eg.: ```2x``` for ```2*x```) but I see it doesn't (it would be ambiguous with hex formats anyway).